### DISCUSSION TITLE
ProgressBar in ContentDialog (WinUI)

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [unoplatform/uno#13675] [Q&A] [answered]
Hey hey :) 

Pretty simple question this time around. I've got a ProgressBar that updates its value over time (say, during a lengthy download). 

If the ProgressBar control is placed in a regular Page, this works just fine : As the value is set, the ProgressBar's visuals are updated. However, when trying the exact same thing inside a ContentDialog, the visuals don't update. Any ideas why this might be 🤔 ?

Thanks

**Comment by limeniye** [comment]
Hello, @BertrandDeSaintBezier.
Thank you for the report!
Can you attache a sample .zip solution document?

**Reply by BertrandDeSaintBezier** [reply]
Okay so while trying to put together a sample for you, I realized that the ProgressBar was declared inside of the ContentDialogue.xaml 
```xaml
<ContentDialog ...>
    <Grid>
        <ProgressBar Maximum="100" Value="{x:Bind Progress, Mode=OneWay}"/>
    </Grid>
</ContentDialog>
```
Instead of being declared as part of the englobing page, as follows :
```xaml
...
<usercontrols:LoginContentDialog>
        <ProgressBar Maximum="100" Value="{x:Bind ViewModel.Progress, Mode=OneWay}"/>
</usercontrols:LoginContentDialog>
...
```
I don't know if this is expected behaviour or not, but it works now.

Can you confirm that this is the way I am supposed to work with the ContentDialog type ?

**Reply by limeniye** [reply, marked as answer]
Hm. To be honest, I guess that its a binding problem.
Are you sure that you did a really working binding at the first sample?

**Reply by BertrandDeSaintBezier** [reply]
Yeah, turns out I was missing a OneWay binding somewhere stupid 🥴

**Reply by limeniye** [reply]
very glad that your problem is solved!
